cmd/stack_create.go

```diff
@@ -0,0 +1,190 @@
+// Copyright © 2019 IBM Corporation and others.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type stackCreateCommandConfig struct {
+	*RootCommandConfig
+	copy string
+}
+
+func newStackCreateCmd(rootConfig *RootCommandConfig) *cobra.Command {
+	config := &stackCreateCommandConfig{RootCommandConfig: rootConfig}
+
+	var stackCmd = &cobra.Command{
+		Use:   "create",
+		Short: "Create a new stack as a copy of an existing stack",
+		Long: `This command will create a new stack as a copy of an existing sample stack in the current directory that has the structure of an Appsody stack.
+        
+		If a copy flag is specified, stack create command will create a new stack as a copy of that existing stack.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) < 1 {
+				return errors.New("Stack create command should have a stack name. Run `appsody stack create <name>` to create one")
+			}
+
+			stack := args[0]
+
+			exists, _ := Exists(stack)
+
+			if exists {
+				return errors.New("This stack named " + stack + " already exists")
+			}
+
+			err := downloadFolderToDisk("https://github.com/appsody/stacks/archive/master.zip", getHome(rootConfig)+"/extract/repo.zip")
+			if err != nil {
+				return err
+			}
+
+			if config.copy != "" {
+				repoIndex := strings.Index(config.copy, "/")
+
+				copiedStack := config.copy[repoIndex+1:]
+
+				valid, unzipErr := unzip(getHome(rootConfig)+"/extract/repo.zip", stack, config.copy)
```

Comment:
All of the places you construct a filesystem path, you need to use `filepath.Join()` so that it works on windows.